src/components/Dashboard/Dashboard.js

```diff
@@ -22,145 +25,183 @@ function Dashboard() {
   // const connectedAccount = useConnectedAccount()
   // const connectedAccount = '0x593e1F9809658d0c92e9f092cF01Aad7D0d734f3'
 
+  const [isAuth, setIsAuth] = useState(true);
+
+  const history = useHistory();
+  const walletAccount = useAccount();
+  useEffect(() => {
+    async function load() {
+      try {
+        // Fetching the password locally.
+        const account = await wallet.login(localStorage.getItem('wpassword'));
+
+        console.log('TESTTTT ACC', account);
+
+        const OrianKey = await localStorage.getItem('orionKey');
+        console.log('OIARKEY', OrianKey);
+
+        if (
+          account === null ||
+          account === undefined ||
+          OrianKey === null ||
+          OrianKey === undefined
+        ) {
+          setIsAuth(false);
+          history.push('/login');
+        }
```

Comment:
Since this is a reusable logic, move it to a separate method of a file in `lib` and may call the file as `Auth.js`.

Reply:
yeah, created a custom Hook for Authentication. it's ready to be merged.